test/views/declaration/CarrierDetailsViewSpec.scala

```diff
@@ -38,482 +37,262 @@ import views.tags.ViewTest
-      messages must haveTranslationFor("declaration.carrier.eori.info")
-      messages must haveTranslationFor("declaration.carrier.address.info")
-    }
+  def createViewWithAddressError(address: Address)(implicit request: JourneyRequest[_]): Document = {
+    val view = createView(
+      CarrierDetails
+        .form(request.declarationType)
+        .fillAndValidate(CarrierDetails(EntityDetails(None, Some(address))))
+    )
+    view must haveGovukGlobalErrorSummary
+    view
   }
 
+  private def createView(form: Form[CarrierDetails])(implicit request: JourneyRequest[_]): Document =
+    carrierDetailsPage(Mode.Normal, form)(request, messages)
+
   "Carrier Details View on empty page" should {
 
-    onJourney(STANDARD, SIMPLIFIED, OCCASIONAL) { implicit request =>
-      val view = createView()
+    "have proper messages for labels" in {
+      val messages = instanceOf[MessagesApi].preferred(journeyRequest())
+      messages must haveTranslationFor("declaration.carrierAddress.title")
+      messages must haveTranslationFor("declaration.address.fullName")
+      messages must haveTranslationFor("declaration.address.fullName.empty")
+      messages must haveTranslationFor("declaration.address.fullName.error")
+      messages must haveTranslationFor("declaration.address.addressLine")
+      messages must haveTranslationFor("declaration.address.addressLine.empty")
+      messages must haveTranslationFor("declaration.address.addressLine.error")
+      messages must haveTranslationFor("declaration.address.townOrCity")
+      messages must haveTranslationFor("declaration.address.townOrCity.empty")
+      messages must haveTranslationFor("declaration.address.townOrCity.error")
+      messages must haveTranslationFor("declaration.address.postCode")
+      messages must haveTranslationFor("declaration.address.postCode.empty")
+      messages must haveTranslationFor("declaration.address.postCode.error")
+      messages must haveTranslationFor("declaration.address.country")
+      messages must haveTranslationFor("declaration.address.country.empty")
+      messages must haveTranslationFor("declaration.address.country.error")
+      messages must haveTranslationFor("site.save_and_continue")
+      messages must haveTranslationFor("site.details.summary_text_this")
+      messages must haveTranslationFor("declaration.carrierAddress.help-item1")
+    }
+
+    onJourney(STANDARD, SIMPLIFIED, OCCASIONAL, CLEARANCE) { implicit request =>
+      val defaultForm = CarrierDetails.form(request.declarationType)
+
       "display page title" in {
 
-        view.getElementById("title") must containMessage("declaration.carrier.title")
+        createView(defaultForm).getElementsByClass("govuk-fieldset__heading").first().text() mustBe messages("declaration.carrierAddress.title")
       }
 
-      "display empty input with label for EORI" in {
+      "display section header" in {
+
+        val view = createView(defaultForm)
```

Comment:
Could this be extracted to not repeat it every test case? Unless you need a new instance every test case.

Reply:
Yes, good spot! Amended